README update (#695)
apple/swift-docc: CONTRIBUTING.md

## Building Swift-DocC

`docc` is the command line interface (CLI) for Swift-DocC and provides

> support for converting and previewing DocC documentation.

Reviewer: ```suggestion
support for generating and previewing documentation.
```

Once you've found an issue to work on,
follow the above instructions for [Building Swift-DocC](#building-swift-docc).

## Technical Overview and Related Projects

> Swift-DocC builds documentation by combining _Symbol Graph_ files containing API information 

Reviewer 2: ```suggestion
Swift-DocC builds documentation by combining _Symbol Graph_ files that contains API information 
```

> with a `.docc` Documentation Catalog containing articles and tutorials

Reviewer 2: ```suggestion
with a `.docc` Documentation Catalog that contains articles and tutorials
```

> to create a final archive containing the compiled documentation.

Reviewer 2: ```suggestion
to create a final archive with the compiled documentation.
```

More concretely, Swift-DocC understands the following kinds of inputs:

 1. _Symbol Graph_ files with the `.symbols.json` extension.
   _Symbol Graph_ files are a machine-readable representation of a module's APIs, 
   including their documentation comments and relationship with one another.

 2. A Documentation Catalog with the `.docc` extension. 
   Documentation Catalogs can include additional documentation content like the following:
  
   - Documentation markup files with the `.md` extension. Documentation markup files can
    be used to extend documentation for symbols and to write free-form articles.
 
   - Tutorial files with the `.tutorial` extension. Tutorial files are used to author
    step-by-step instructions on how to use a framework.
 

>    - Additional documentation assets with known extensions like `.png`, `.jpg`, `.mov`,

Author: @ethan-kusters This is the closest doc that I found https://www.swift.org/documentation/docc/formatting-your-documentation-content#Add-Images-to-Your-Content, but needs to be updated with the support for `.mov`.

- [`SwiftDocCUtilities` framework documentation](https://apple.github.io/swift-docc/documentation/swiftdoccutilities/)

### Related Projects

 - As of Swift 5.5, the [Swift Compiler](https://github.com/apple/swift) is able to 
  emit _Symbol Graph_ files as part of the compilation process.
   
 - [SymbolKit](https://github.com/apple/swift-docc-symbolkit) is a Swift package containing
  the specification and reference model for the _Symbol Graph_ File Format.
  
 - [Swift Markdown](https://github.com/apple/swift-markdown) is a 
  Swift package for parsing, building, editing, and analyzing 
  Markdown documents. It includes support for the Block Directive elements
  that Swift-DocC's tutorial files rely on.
   
 - [Swift-DocC-Render](https://github.com/apple/swift-docc-render) 
  is a web application that understands and renders
  Swift-DocC's _render JSON_ format.
   
 - [Xcode](https://developer.apple.com/xcode/) consists of a suite of
  tools that developers use to build apps for Apple platforms.
  Beginning with Xcode 13, Swift-DocC is integrated into Xcode
  with support for building and viewing documentation for your framework and
  its dependencies.

> <!-- Copyright (c) 2021-2022 Apple Inc and the Swift Project authors. All Rights Reserved. -->

Reviewer 3: ```suggestion
<!-- Copyright (c) 2021-2023 Apple Inc and the Swift Project authors. All Rights Reserved. -->
```
